src/ds/logger.h

```diff
@@ -137,7 +137,8 @@ namespace logger
       if (len > max_len)
         data += len - max_len;
 
-      ss << fmt::format("[{:<5}] {:<36} | ", config::to_string(ll), data);
+      ss << fmt::format(
+        "status=[{:<5}] loc={:<36} | msg=", config::to_string(ll), data);
```

Comment:
I'd prefer not to alter the log format here - I think it makes the normal logs much less readable. Since the Python infrastructure already needs to know whether it is forwarding to an analytics log file, can we instead parse and reformat the lines in Python?

Reply:
Ah obviously this isn't as easy as I'd thought - the process's stdout needs to be a file descriptor, so plugging in a Python translation layer is not so trivial. Hmmmm.

Reply:
If we keep the keys identical, we could pass in the format through an env var. We could have some predefined values « default », « azure logging », and then allow

Reply:
This formatting is just adding delimiters so the [parse](https://docs.microsoft.com/en-us/azure/azure-monitor/log-query/parse-text) task can use a simple pattern:

```
parse RawData with * "time=" eventtime " status=" status " loc=" loc "msg=" msg
```

I think we already have sufficient delimiters to parse this with the existing format:

```
2019-08-21 15:24:59.687834        [info ] ../CCF/src/host/sigterm.h:24         | SIGTERM: Shutting down enclave gracefully...

parse RawData with * eventtime "[" status "]" loc "|" msg
```

and this makes it clear that the log is dynamically parseable to whatever level the user wants:

```
parse rawData with * prefix "|" msg
parse rawData with * year "-" month "-" day " " time "[" status "]" file "." fileextension ":" linenumber "|" message
```

Either approach breaks down with unexpected changes ("|" added to the format, "msg=" in a filename), but the key=value approach implies stronger structural protection than is actually provided.

Reply:
I'll leave the logs in their default state for now and we can come back and look at it later if we think it's not sufficient. I do like the idea of having the user define the log format however so I will think about it.